src/components/utils/api.js

```diff
@@ -19,7 +19,85 @@ export const getCommentsByArticleId = (id) => {
 };
 
 export const patchVote = (path, vote) => {
-  return articlesAPI
-    .patch(path, { inc_votes: vote })
-    .catch((err) => console.log(err));
+  return articlesAPI.patch(path, { inc_votes: vote });
+};
+
+export const errorTimeout = ({
+  setDoubleClickUp,
+  setDoubleClickDn,
+  setError,
+}) => {
+  setTimeout(() => {
+    setDoubleClickUp(false);
+    setDoubleClickDn(false);
+    setError(null);
+  }, 2000);
+};
+
+export const changeVote = (
+  e,
+  path,
+  setVotes,
+  setPreviousClick,
+  previousClick,
+  setDoubleClickUp,
+  setDoubleClickDn,
+  setError
+) => {
+  e.preventDefault();
+  const {
+    target: { value },
+  } = e;
+  setVotes((prevVotes) => prevVotes + +value);
+  setPreviousClick((prePre) => +prePre + +value);
+  if (value > 0 && previousClick === 0) {
+    (() => {
+      setDoubleClickUp(true);
+      setDoubleClickDn(false);
+    })();
+  }
+  if (value > 0 && previousClick === -1) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(false);
+    })();
+  }
+  if (value < 0 && previousClick === 0) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(true);
+    })();
+  }
+  if (value < 0 && previousClick === 1) {
+    (() => {
+      setDoubleClickUp(false);
+      setDoubleClickDn(false);
+    })();
```

Comment:
This code does the job but repeats itself a bit, could benefit from a refactor in the future, it's also not relevant for this ticket, and should be included in the vote on an article ticket instead
It's best practise to limit functionality to what is needed to meet the needs of one ticket at a time